v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapper.java

```diff
@@ -156,8 +158,15 @@ static Boolean avroFieldToBoolean(Object recordValue, Schema fieldSchema) {
     if (recordValue == null) {
       return null;
     }
+    String val = recordValue.toString();
```

Comment:
What would happen if we skip this stringification, and just cast the object to a java boolean?
When we stringify, there is a possibility that the underlying library might return "0" or "false" (similarly "1" or "true"), 
For the documentation, it looks like avro's  boolean stringifies to "0" and "1" and avro's boolean within a json (does not affect this part of the code) stringifies to "false" or "true"


If we depend on the avro stringification (no harm), and why not just return `val.equals("1")`?

```
 if (Arrays.asList("0", "1").contains(val)) {
      return val.equals("1");
    }
    // BooleanUtils.toBoolean() never throws exception, so we don't need to catch it.
    return BooleanUtils.toBoolean(val);
    ```
    Makes some one not so familier (like me) feel there's some kind of undefined behavior in the last return.

Reply:
Typecasting to boolean works for boolean source types only. Converting it to string then boolean allows handling long, string, boolean, char et. This stringification handles more cases.

`why not just return val.equals("1")` We could do that also, but boolean utils also handle cases like true, false, t, f etc. 

The combination of the 2 means if we can infer a 0 or 1 use that to decide else rely on boolean utils.

I get it is not intuitive but it supports alot of cases this way. Alot of these cases might be unnecessary cases as well, but did it because its a small effort to cover them.

I'll add a detailed clarifying comment if we decide to keep it this way.